components/script/dom/xrrigidtransform.rs

```diff
@@ -134,15 +134,11 @@ impl XRRigidTransformMethods for XRRigidTransform {
         })
     }
     // https://immersive-web.github.io/webxr/#dom-xrrigidtransform-matrix
-    fn Matrix(&self, _cx: JSContext) -> NonNull<JSObject> {
-        if self.matrix.get().is_null() {
-            let cx = GlobalScope::get_cx();
-            // According to the spec all matrices are column-major,
-            // however euclid uses row vectors so we use .to_array()
-            let arr = self.transform.to_transform().to_array();
-            create_typed_array(cx, &arr, &self.matrix);
+    fn Matrix(&self, _cx: JSContext) -> Float32Array {
+        match self.matrix.get_internal() {
+            Ok(matrix) => matrix,
+            Err(_) => todo!(),
```

Comment:
@gterzian what should we return in this case?

Reply:
What you want to do is: in the `!self.matrix.is_initialized` branch, use `set_data` to initialize the float 32 array to the data of `self.transform`, and only then call `self.matrix.get_internal().expect`